src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java

```diff
@@ -76,8 +77,18 @@ public UserTransaction assignHearingPartToSessionWithTransaction(UUID hearingPar
                                                                      HearingPartSessionRelationship assignment)
                                                                         throws IOException {
         HearingPart hearingPart = hearingPartRepository.findOne(hearingPartId);
+        if(hearingPart.getVersion() != assignment.getHearingPartVersion()) {
```

Comment:
no, we want to use the existing mechanism not to implement it in many places, use detach instead, so the standard logic will be used.

Reply:
fixed